python/pyspark/pandas/tests/test_groupby.py

```diff
@@ -1380,8 +1380,18 @@ def test_nth(self):
         for n in [0, 1, 2, 128, -1, -2, -128]:
             self._test_stat_func(lambda groupby_obj: groupby_obj.nth(n))
 
+        pdf, psdf = self.pdf, self.psdf
         with self.assertRaisesRegex(TypeError, "Unsupported type"):
-            self.psdf.groupby("B").nth("x")
+            psdf.groupby("B").nth("x")
+
+        # in case there is no aggregation columns, the results should be the same
+        if LooseVersion("1.4.0") <= LooseVersion(pd.__version__) < LooseVersion("1.5.0"):
```

Comment:
qq: So, do we need upperbound (1.5.0) here since we're going to only support pandas 1.4.x for the Apache Spark 3.4.0 ?

Reply:
`< LooseVersion("1.5.0")`
I think only `1.4.x` will test this case